README.md

```diff
@@ -103,33 +114,63 @@ easy_install --upgrade "ibmcloudant>=0.0.27"
-    1. [*Basic* (or legacy) *authentication*](#basic-authentication) is a fallback for both [Cloudant](https://cloud.ibm.com/docs/services/Cloudant/api?topic=cloudant-authentication#basic-authentication) and [Apache CouchDB](https://docs.couchdb.org/en/stable/api/server/authn.html#basic-authentication) back-end database servers. This authentication type requires the good old `username` and `password` credentials.
-    1. *Noauth* authentication does not need any credentials. Note that this authentication type will only work for queries against a database with read access for everyone.
-1. The service `url`
+This library requires some of your
+[Cloudant service credentials][service-credentials] to authenticate with your
+account.
 
-You have to add these properties as your **environment variables**, because some
-examples that follow assume that these variables are set.
-To learn more about authentication configuration see the related documentation in the
-[Cloudant API docs](https://cloud.ibm.com/apidocs/cloudant#authentication?code=python) or in the
-[general SDK usage information](https://github.com/IBM/ibm-cloud-sdk-common#authentication).
+1. `IAM`, `COUCHDB_SESSION`, `BASIC` or `NOAUTH` **authentication type**.
+1. [*IAM authentication*](#iam-authentication) is highly recommended when your
+back-end database server is [**Cloudant**][cloud-IAM-mgmt]. This
+authentication type requires a server-generated `apikey` instead of a
+user-given password.
+1. [*Session cookie (`COUCHDB_SESSION`) authentication*](#session-cookie-authentication)
+is recommended for [Apache CouchDB][couch-cookie-auth] or for
+[Cloudant][cloudant-cookie-auth] when IAM is unavailable. It exchanges username
+and password credentials for an `AuthSession` cookie from the `/_session`
+endpoint.
+1. [*Basic* (or legacy) *authentication*](#basic-authentication) is a fallback
+for both [Cloudant][cloudant-basic-auth] and [Apache CouchDB][couch-basic-auth]
+back-end database servers. This authentication type requires the good old
+`username` and `password` credentials.
+1. *Noauth* authentication does not need any credentials. Note that this
+authentication type will only work for queries against a database with read
+access for everyone.
+1. The service `url`.
+
+You can add these properties as your
+[environment variables](#authenticate-with-environment-variables), but
+if you decide to set these properties
+[programatically](#authenticate-with-external-configuration)
+or
+[with the help of an external credentials file](#authenticate-with-external-configuration),
+please check the linked sections. The code examples below will follow the
+[authentication with environment variables](#authenticate-with-environment-variables).
```

Comment:
What about creating a list for the possible ways:
```suggestion
There are several ways to use these properties:
1. As [environment variables](#authenticate-with-environment-variables)
2. The [programatic approach](#authentication-programatically)
3. [External credentials file](#authenticate-with-external-configuration)

The code examples below will follow the
[authentication with environment variables](#authenticate-with-environment-variables).
```

Reply:
I like this suggested rephrase with a teeny-tiny-minor request:
`There are several ways to **set** these properties:`

Reply:
It is in eec751f.